src/ILLink.RoslynAnalyzer/TrimAnalysis/ReflectionAccessAnalyzer.cs

```diff
@@ -0,0 +1,96 @@
+					break;
+				*/
+				case IEventSymbol @event:
+					GetDiagnosticsForEvent (diagnosticContext, @event);
+					break;
+				}
+			}
+		}
+
+		static void ReportRequiresUnreferencedCodeDiagnostic (in DiagnosticContext diagnosticContext, AttributeData requiresAttributeData, ISymbol member)
+		{
+			var message = RequiresUnreferencedCodeUtils.GetMessageFromAttribute (requiresAttributeData);
+			var url = RequiresAnalyzerBase.GetUrlFromAttribute (requiresAttributeData);
+			diagnosticContext.AddDiagnostic (DiagnosticId.RequiresUnreferencedCode, member.GetDisplayName (), message, url);
+		}
+
+		static void GetDiagnosticsForMethod (in DiagnosticContext diagnosticContext, IMethodSymbol methodSymbol)
+		{
+			if (methodSymbol.TryGetRequiresUnreferencedCodeAttribute (out var requiresAttributeData))
+				ReportRequiresUnreferencedCodeDiagnostic (diagnosticContext, requiresAttributeData, methodSymbol);
+
+			if (!methodSymbol.IsStatic && methodSymbol.GetDynamicallyAccessedMemberTypes () != DynamicallyAccessedMemberTypes.None)
+				diagnosticContext.AddDiagnostic (DiagnosticId.DynamicallyAccessedMembersMethodAccessedViaReflection, methodSymbol.GetDisplayName ());
+			else if (methodSymbol.IsVirtual && FlowAnnotations.GetMethodReturnValueAnnotation (methodSymbol) != DynamicallyAccessedMemberTypes.None)
+				diagnosticContext.AddDiagnostic (DiagnosticId.DynamicallyAccessedMembersMethodAccessedViaReflection, methodSymbol.GetDisplayName ());
+			else {
+				foreach (var parameter in methodSymbol.Parameters) {
+					if (FlowAnnotations.GetMethodParameterAnnotation (parameter) != DynamicallyAccessedMemberTypes.None) {
+						diagnosticContext.AddDiagnostic (DiagnosticId.DynamicallyAccessedMembersMethodAccessedViaReflection, methodSymbol.GetDisplayName ());
+						break;
+					}
+				}
+			}
+		}
+
+		static void GetDiagnosticsForProperty (in DiagnosticContext diagnosticContext, IPropertySymbol propertySymbol)
+		{
+			if (propertySymbol.SetMethod is not null)
+				GetDiagnosticsForMethod (diagnosticContext, propertySymbol.SetMethod);
+			if (propertySymbol.GetMethod is not null)
+				GetDiagnosticsForMethod (diagnosticContext, propertySymbol.GetMethod);
+		}
+
+		static void GetDiagnosticsForEvent (in DiagnosticContext diagnosticContext, IEventSymbol eventSymbol)
+		{
+			if (eventSymbol.AddMethod is not null)
+				GetDiagnosticsForMethod (diagnosticContext, eventSymbol.AddMethod);
+			if (eventSymbol.RemoveMethod is not null)
+				GetDiagnosticsForMethod (diagnosticContext, eventSymbol.RemoveMethod);
+			if (eventSymbol.RaiseMethod is not null)
+				GetDiagnosticsForMethod (diagnosticContext, eventSymbol.RaiseMethod);
+		}
+
+		static void GetDiagnosticsForField (in DiagnosticContext diagnosticContext, IFieldSymbol fieldSymbol)
+		{
+			if (fieldSymbol.TryGetRequiresUnreferencedCodeAttribute (out var requiresAttributeData))
+				ReportRequiresUnreferencedCodeDiagnostic (diagnosticContext, requiresAttributeData, fieldSymbol);
```

Comment:
I don't think we allows RUC on a field, right?

Reply:
No, we don't - deleted the code

Reply:
Apparently, we are when RequiresUnreferencedCode is on a class, I think we mark access to fields as dangerous because they can call the static constructor. Getting back the code